Library/brew.sh

```diff
@@ -57,7 +57,7 @@ then
   unset HOMEBREW_RUBY_PATH
 fi
 
-HOMEBREW_SYSTEM="$(uname -s)"
+HOMEBREW_SYSTEM="$(/usr/bin/uname -s)"
```

Comment:
I think we may relax full path for `uname`. Otherwise, it would just make brew unrunable for certain Linux distro.

Alternatively, we could do something like:
```bash
uname()
{
   [[ -x "/usr/bin/uname" ]] && { /usr/bin/uname "$@"; return; }
   [[ -x "/bin/uname" ]] && { /bin/uname "$@"; return;  }
}
```

What do you think?